buildrunner/docker/runner.py

```diff
@@ -251,6 +254,94 @@ def cleanup(self):
+                    print(f"Cache directory {cache_dir} does not exist, "
+                          f"skipping restore of archive {local_cache_archive_file}")
+                    continue
+
+                files = [f for f in listdir(cache_dir) if isfile(join(cache_dir, f))]
+
+                cache_key = local_cache_archive_file\
+                    .replace(f"{cache_dir}/", "")\
+                    .replace(f".{cache_archive_ext}", "")
+
+                most_recent_time = 0
+                local_cache_archive_match = None
+                for file in files:
+                    if file.startswith(cache_key):
+                        print(f"    starts with {cache_key}")
+                        curr_archive_file = join(cache_dir, file)
+                        mod_time = getmtime(curr_archive_file)
+                        if mod_time > most_recent_time:
+                            most_recent_time = mod_time
+                            local_cache_archive_match = curr_archive_file
+
+                if local_cache_archive_match is None:
+                    print(f"WARNING: Not able to restore cache {docker_path} since "
+                          f"there was no prefix matching for [{local_cache_archive_file}]")
+                    continue
+
+                local_cache_archive_file = local_cache_archive_match
+
+            orig_shell = self.shell
+            try:
+                self.shell = "/bin/sh"
+                exit_code = self.run(f"mkdir -p {docker_path}")
+                if exit_code:
+                    print(f"WARNING: There was an issue creating {docker_path} on the docker container.")
+
+                with open(local_cache_archive_file, 'rb') as data:
+                    print(f"{local_cache_archive_file}:{docker_path}")
+
+                    restored_cache_src.add(docker_path)
+                    if not self.docker_client.put_archive(self.container['Id'], docker_path, data):
+                        print(f"WARNING: An error occurred when trying to use cache "
+                              f"{local_cache_archive_file}:{docker_path}")
+
+            except docker.errors.APIError as exception:
+                print(f"WARNING: An docker.errors.APIError has occurred\n{exception}")
+            finally:
+                self.shell = orig_shell
+
+    def save_caches(self, caches: OrderedDict):
+        """
+        Saves caches from a source locations in the docker container to locations on the host system as archive file.
+        """
+        saved_cache_src = set()
+        if caches and isinstance(caches, OrderedDict):
+            for local_cache_archive_file, docker_path in caches.items():
+                if docker_path not in saved_cache_src:
+                    saved_cache_src.add(docker_path)
+                    print(f"Saving cache [{docker_path}] "
+                          f"running on container {self.container['Id']} "
+                          f"to local cache [{local_cache_archive_file}]")
```

Comment:
Why the square brackets here? This makes me think it's a list, but it's not...

Reply:
No reason, I was just being inconsistent here. I'll add ticks like in the other documentation that I'll update.

- [ ] Replace `[]` with ticks